components/shared/compositing/display_list.rs

```diff
@@ -299,6 +413,27 @@ impl ScrollTreeNode {
             },
         };
     }
+
+    fn invalidate_cached_transforms(&self, scroll_tree: &ScrollTree, ancestors_invalid: bool) {
+        let node_invalid = match &self.info {
+            SpatialTreeNodeInfo::Scroll(info) if info.offset_changed.get() => {
+                info.offset_changed.set(false);
```

Comment:
Maybe simpler with `take()`

```suggestion
            SpatialTreeNodeInfo::Scroll(info) if info.offset_changed.take() => {
```